components/NavigationBar/examples/NavigationBarIconsExample.m

```diff
@@ -78,19 +78,21 @@ - (void)viewDidLoad {
   self.navigationItem.rightBarButtonItem = trailingButtonItem;
   self.navigationItem.backBarButtonItem = backButtonItem;
 
-  if ([self.view respondsToSelector:@selector(safeAreaLayoutGuide)]) {
-    UILayoutGuide *layoutGuide = [self.view performSelector:@selector(safeAreaLayoutGuide)];
-    [layoutGuide.topAnchor constraintEqualToAnchor:self.navigationBar.topAnchor].active = YES;
-  } else {
-    [NSLayoutConstraint constraintWithItem:self.topLayoutGuide
-                                 attribute:NSLayoutAttributeBottom
-                                 relatedBy:NSLayoutRelationEqual
-                                    toItem:self.navigationBar
-                                 attribute:NSLayoutAttributeTop
-                                multiplier:1.0
-                                  constant:0]
-        .active = YES;
-  }
+  if ([[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:(NSOperatingSystemVersion){
```

Comment:
This should be protected with a compile-time SDK check: 

```objectivec


#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
  if (@available(iOS 11.0, *)) {
    // iOS 11 code goes here
  } else {
#endif

  // iOS 8-10 code goes here

#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
  } // iOS 11 SDK
#endif
```

Reply:
Thanks @romoore will update accordingly.